plugins/scaffolder-react/src/next/components/ReviewState/util.ts

```diff
@@ -20,8 +20,11 @@ export function isJsonObject(value?: JsonValue): value is JsonObject {
   return typeof value === 'object' && value !== null && !Array.isArray(value);
 }
 
-// Helper function to get the last part of the key
-export function getLastKey(key: string): string {
+// Helper function to format a key into a human-readable string
+export function formatKey(key: string): string {
   const parts = key.split('/');
-  return parts[parts.length - 1];
+  return parts
+    .filter(Boolean)
+    .map(part => part.charAt(0).toLocaleUpperCase('en-US') + part.slice(1))
```

Comment:
Using `lodash/startCase` may work better here (it's used in other parts of scaffolder UI for similar formatting)

Reply:
`startCase` is a bit more opinionated (e.g. removes special characters), and YAML accepting special chars in quoted keys means I think we should still support this - so to retain the [current behaviour](https://github.com/backstage/backstage/pull/26140/files#diff-9d5518eed3fe4ace9a53f765ca0f2865d485c7f2c6a76bd8bb0220ae9abe3337R77) I went for this manual capitalisation instead!

Assuming we'd want to display the full key (including special chars) in the output table, I think this might be necessary 🤔

Reply:
Understood. `startCase` currently is the default title format for the review page. [See current code here](https://github.com/backstage/backstage/blob/f54d3a5385659eebb7e629c7f2ab8e763c67a111/packages/core-components/src/components/StructuredMetadataTable/StructuredMetadataTable.tsx#L179). Imo it would make sense  to keep using that formatting for consistency and remove test case for special chars if needed but I'll let the maintainers weigh in on this.

Reply:
@richcooper95, yeah, kinda leaning towards using the existing formatting solution that we use indirectly already.

Think that the special start character one is a little bit of an edge case, and think we can take that separately if needed.

Reply:
Cool, sounds good! Just changed in https://github.com/backstage/backstage/pull/26140/commits/23ac83be20198321087170177cdf2b064936efb0 👍